scripts/scripts.js

```diff
@@ -90,9 +90,65 @@ sampleRUM.observe = ((elements) => {
   });
 });
 
+sampleRUM.targetselector = (element) => {
+  let selector = element.tagName.toLowerCase();
+  // always include these attributes as they have valuable info
+  const distinctAttributes = ['src', 'href'];
+  // capture the actual click target instead of a child element
+  const terminateearly = (e) => e.tagName.toLowerCase() === 'a'
+    || e.tagName.toLowerCase() === 'button';
+  if (element === document.body || element === document.documentElement) {
+    return selector;
+  }
+  if (element.id) {
+    return `#${element.id}`;
+  }
+  if (element.className) {
+    selector = `${selector}.${Array.from(element.classList).join('.')}`;
+  }
+  selector = distinctAttributes.reduce((pv, attname) => {
+    if (element.getAttribute(attname)) {
+      return `${pv}[${attname}="${element.getAttribute(attname)}"]`;
+    }
+    return pv;
+  }, selector);
+  // check for uniqueness
+  const otherchildren = element.parentElement.querySelectorAll(element.tagName);
+  for (let i = 0; (
+    i < otherchildren.length
+    && otherchildren.length > 1
+    && element.parentElement.querySelectorAll(selector).length > 1);
+    i += 1) {
+    if (otherchildren[i] === element) {
+      selector = `${element.tagName.toLowerCase()}:nth-of-type(${i + 1})`;
+      break;
+    }
+  }
+  // recurse
+  const parent = sampleRUM.targetselector(element.parentElement);
+
+  let ancestor = element.parentElement;
+  while (ancestor) {
+    if (terminateearly(ancestor)) {
```

Comment:
Point for discussion: do we want to track *every* click, even on elements that are non-interactive? At the moment, we track every click, but if you click something within an interactive element, we just use the interactive ancestor instead of adding the path to the `div > p > img` that has actually been clicked.